core/src/main/java/feign/AsyncFeign.java

```diff
@@ -255,24 +267,27 @@ protected AsyncFeign(AsyncBuilder<C> asyncBuilder) {
     this.activeContext = new ThreadLocal<>();
 
     this.defaultContextSupplier = asyncBuilder.defaultContextSupplier;
-    this.client = asyncBuilder.client;
 
-    this.logLevel = asyncBuilder.logLevel;
-    this.logger = asyncBuilder.logger;
+    this.client = Capability.enrich(asyncBuilder.client, asyncBuilder.capabilities);
+    this.logLevel = Capability.enrich(asyncBuilder.logLevel, asyncBuilder.capabilities);
+    this.logger = Capability.enrich(asyncBuilder.logger, asyncBuilder.capabilities);
+    Decoder decoder = Capability.enrich(asyncBuilder.decoder, asyncBuilder.capabilities);
 
     this.responseHandler = new AsyncResponseHandler(
-        asyncBuilder.logLevel,
-        asyncBuilder.logger,
-        asyncBuilder.decoder,
+        this.logLevel,
+        this.logger,
+        decoder,
         asyncBuilder.errorDecoder,
         asyncBuilder.dismiss404,
         asyncBuilder.closeAfterDecode);
 
-    asyncBuilder.builder.client(this::stageExecution);
-    asyncBuilder.builder.decoder(this::stageDecode);
-    asyncBuilder.builder.forceDecoding(); // force all handling through stageDecode
-
-    this.feign = asyncBuilder.builder.build();
+    this.feign = asyncBuilder.builder
+        .client(this::stageExecution)
+        .skipClientEnrichment() // don't enrich the wrapper
```

Comment:
I don't like this trend...  what if we only enrich here, not on the other builder?

Reply:
That's what these flag functions do - they tell that other builder not to enrich because we did it here. These three flags essentially do the same - that `AsyncFeign` will work differently. The problem is that I don't want to tie them to "forceDecoding" as it's shady logic. So I need something else. I might combine both "skip" flags into one.